src/ape/managers/compilers.py

```diff
@@ -179,6 +180,37 @@ def compile(self, contract_filepaths: List[Path]) -> Dict[str, ContractType]:
 
         return contract_types_dict
 
+    def compile_source(
+        self,
+        compiler_name: str,
+        code: str,
+        settings: Optional[Dict] = None,
+        contract_type_overrides: Optional[Dict] = None,
+    ) -> ContractContainer:
+        """
+        Compile the given program.
+
+        Usage example::
+
+            code = '[{"name":"foo","type":"fallback", "stateMutability":"nonpayable"}]'
+            contract_type = compilers.compile_source("ethpm", code, contractName="MyContract")
```

Comment:
like this actually, or this needs to change

Reply:
this is the manager, not compiler API, but ya this doc is wrong now

Reply:
so manager i didnt want this because it is not clear which dict gets the **, the settings or the overrides, so it seems nicer to just have 2 dicts are optional args.

in general, i tend to prefer this anyway to avoid issue mentioned here: https://github.com/ApeWorX/ape/pull/1707#discussion_r1376633835